### DISCUSSION TITLE
Docs for associated refinements

### DISCUSSION THREAD

**Discussion by ranjitjhala** [flux-rs/flux#929] [General]
Per @enjhnsn2 's suggestion -- I agree we'd want to have some docs on associated refinements; what is a good progression? 

Is there something simpler than the iterator/vec-slice-index/ that folks have in mind?

**Comment by nilehmann** [comment]
I think implementing `Index` for a custom container is a good starting point (e.g. `RVec`) so you don't need to deal with the various indirections of the `Vec` implementation.

**Reply by nilehmann** [reply]
Hmm, but that requires modifying an external definition (adding an assoc refinement to Index). It would be good to have an example where everything is local
